src/linear_algebra/clifford_algebra/equivs.lean

```diff
@@ -0,0 +1,89 @@
+/-
+Copyright (c) 2021 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import linear_algebra.clifford_algebra.basic
+import data.complex.module
+
+/-!
+# Other constructions isomorphic to Clifford Algebras
+
+This file contains isomorphisms showing that other types are equivalent to some `clifford_algebra`:
+
+* `clifford_algebra_ring.equiv`: any ring is equivalent to a `clifford_algebra` over an empty
+  vector space.
+* `clifford_algebra_complex.equiv`: the `complex` numbers are equivalent to a
+  `clifford_algebra` over a one-dimensional vector space with a quadratic form that satisfies
+  `Q (ι Q 1) = -1`.
+
+Future work:
+
+* Isomorphism to the `quaternion`s over `ℝ × ℝ`, sending `i, j, k` to `(0, 1)`, `(1, 0)`, and
+  `(1, 1)` (port from the `lean-ga` project).
+-/
+
+open clifford_algebra
+
+namespace clifford_algebra_ring
+
+variables {R : Type*} [comm_ring R]
+
+@[simp]
+lemma ι_eq_0 : ι (0 : quadratic_form R unit) = 0 :=
+subsingleton.elim _ _
+
+/-- The clifford algebra over a 0-dimensional vector space is isomorphic to its scalars. -/
+def equiv : clifford_algebra (0 : quadratic_form R unit) ≃ₐ[R] R :=
+alg_equiv.of_alg_hom
+  (clifford_algebra.lift (0 : quadratic_form R unit) $
+    ⟨0, λ m : unit, (zero_mul (0 : R)).trans (algebra_map R _).map_zero.symm⟩)
+  (algebra.of_id R _)
+  (by { ext x, exact (alg_hom.commutes _ x), })
+  (by { ext : 1, rw [ι_eq_0, linear_map.comp_zero, linear_map.comp_zero], })
+
+end clifford_algebra_ring
+
+namespace clifford_algebra_complex
+
+/-- The quadratic form sending elements to the negation of their square. -/
+def Q : quadratic_form ℝ ℝ :=
+-quadratic_form.lin_mul_lin linear_map.id linear_map.id
```

Comment:
Is `Q` a commonly used name for this map, or just an easy placeholder? In the latter case, I would prefer either the name `neg_sq`, or something like making it a `protected` def. (Otherwise, I foresee confusion between the def `Q` and the variable `Q` that a lot of literature uses for a generic quadratic form, or between `ℚ` and `Q`.)

Reply:
No, I only called it this because it's the name of the argument to `clifford_algebra`.

My only resistance to renaming it `neg_sq` is I don't know what analogous name I'd use for the quaternions - but certainly making it protected would be sensible.

It should certainly be protected though

Reply:
Would something like `smul_sq_add_smul_sq` work for the quaternion case? (Or even `smul_sq` if the previous would be too verbose?) Otherwise, I'd go for `protected`.

Reply:
```suggestion
protected def Q : quadratic_form ℝ ℝ :=
-quadratic_form.lin_mul_lin linear_map.id linear_map.id
```

Reply:
Ok, I finally came back to this - I couldn't make it protected immediately as that made everything that followed super verbose, but I did `attribute [protected]` at the end of the section.